### PR TITLE
remove dead code udp_tracker_connection::pick_target_endpoint

### PR DESCRIPTION
**Pull request by ssiloti**
Endpoints are filtered by address family in name_lookup() so this
function will always return the first endpoint.

### CODE REVIEW

File: src/udp_tracker_connection.cpp [hunk 1]
```diff
@@ -154,7 +154,7 @@ namespace libtorrent {
 #endif
 
 		// pick another target endpoint and try again
-		m_target = pick_target_endpoint();
+		m_target = udp::endpoint(m_endpoints.front().address(), m_endpoints.front().port());
```
Comment (arvidn) [inline src/udp_tracker_connection.cpp]: i seem to recall there is a utility function already for converting between udp and tcp endpoints.

File: src/udp_tracker_connection.cpp [hunk 2]
```diff
@@ -252,49 +252,11 @@ namespace libtorrent {
 			return;
 		}
 
-		m_target = pick_target_endpoint();
```
Comment (arvidn) [inline src/udp_tracker_connection.cpp]: `pick_target_endpoint()` seems to support the case `m_endpoints.empty()` as well. are you sure that can't happen?
there should probably be an assert for that.

Comment (ssiloti) [reply src/udp_tracker_connection.cpp]: On the contrary, `pick_target_endpoint` assumes `!m_endpoints.empty()` by unconditionally dereferencing `m_endpoints.begin()` on its second line.

There is a check just above this line which returns if `m_endpoints.empty()` and a similar one above other other callsite so neither line can be reached if `m_endpoints.empty()`.

File: src/udp_tracker_connection.cpp (hunk 1 above)
Comment (ssiloti) [reply src/udp_tracker_connection.cpp]: I cannot find such a function.

**Pull request merged**